static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,86 @@
+import {useCallback, useContext, useEffect} from 'react';
+import styled from '@emotion/styled';
+
+import {DrawerBody, DrawerPanel} from 'sentry/components/globalDrawer/components';
+import {DrawerContext} from 'sentry/components/globalDrawer/context';
+import {useLocation} from 'sentry/utils/useLocation';
+
+export interface DrawerOptions {
+  //
+  // If true (default), closes the drawer when an escape key is pressed
+  //
+  closeOnEscapeKeypress?: boolean;
+  //
+  // If true (default), closes the drawer when anywhere else is clicked
+  //
+  closeOnOutsideClick?: boolean;
+  //
+  // Callback for when the drawer closes
+  //
+  onClose?: () => void;
+  //
+  // Callback for when the drawer opens
+  //
+  onOpen?: () => void;
+}
+
+export interface DrawerRenderProps {
+  //
+  // Body container for the drawer
+  //
+  Body: typeof DrawerBody;
+  //
+  // Close the drawer
+  //
+  closeDrawer: () => void;
+}
+
+interface GlobalDrawerProps {}
+
+export default function GlobalDrawer(_props: GlobalDrawerProps) {
+  const location = useLocation();
+  const {config, closeDrawer: ctxCloseDrawer} = useContext(DrawerContext);
+  const {renderer, options = {}} = config;
+  const {closeOnEscapeKeypress = true, closeOnOutsideClick = true} = options;
+
+  const closeDrawer = useCallback(() => {
+    // Callsite callback when closing the drawer
+    options?.onClose?.();
+    // Actually close the drawer component
+    ctxCloseDrawer();
+  }, [options, ctxCloseDrawer]);
+
+  // Close the drawer when the browser history changes.
+  useEffect(() => ctxCloseDrawer(), [location.pathname, ctxCloseDrawer]);
+
+  const isDrawerOpen = typeof renderer === 'function';
+  const renderedChild =
+    renderer?.({
+      Body: DrawerBody,
+      closeDrawer,
+    }) ?? null;
+
+  return (
+    <DrawerContainer data-test-id="drawer-container">
```

Comment:
Possible to not have a data-test-id here? What are we using this for. Ideally we should avoid using `data-test-id`

See https://testing-library.com/docs/queries/bytestid/

> In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post ["Making your UI tests resilient to change"](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)

Reply:
Moved all usage of `data-test-id` to the fake elements added in tests :+1: